src/classes/Addresses.cls

```diff
@@ -273,14 +270,15 @@ public inherited sharing class Addresses extends fflib_SObjects {
             }
 
             // update HH Account's Billing address fields
-            Account acc = new Account(Id = accId);
-            NPSP_HouseholdAccount npspHouseholdAccount = new NPSP_HouseholdAccount
-                    (accountsById.get(accId));
             NPSP_Address npspAddress = new NPSP_Address(addr);
+
+            NPSP_HouseholdAccount npspHouseholdAccount = new NPSP_HouseholdAccount(accountsById.get(accId));
+            npspHouseholdAccount.isBillingAddressUndeliverable(npspAddress.isUndeliverable());
+
             if (npspHouseholdAccount.billingAddressIsDifferentFrom(npspAddress)) {
                 npspHouseholdAccount.billingAddress(npspAddress);
-                listAccUpdate.add(npspHouseholdAccount.getRecord());
             }
+            listAccUpdate.add(npspHouseholdAccount.getRecord());
```

Comment:
Is there really no condition to check for this now? It seems like a lot more Accounts will get updated outside of that `if` block

Reply:
Yeah I had that on the to-do list to make that filter more specific. It's updated now.